dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/ToolResponseResult.java

```diff
@@ -11,17 +11,46 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
+
+import jakarta.json.bind.Jsonb;
 
 /**
  *
  */
 public class ToolResponseResult {
     private boolean isError = false;
+    private Object structuredContent;
 
-    public ToolResponseResult(Object result, boolean isError) {
+    public ToolResponseResult(Object result, Jsonb jsonb, boolean isError) {
         this.isError = isError;
+        if (isPojo(result)) {
+            content.add(new TextContent(jsonb.toJson(result)));
+            structuredContent = result;
+        } else {
+            content.add(new TextContent(Objects.toString(result)));
+        }
```

Comment:
I think we should flip this logic around.

If an object is of a type that we know how to deal with (e.g. Strings, primitives, wrapper types at present), we should process them with that logic, otherwise we should treat it as a pojo and allow jsonb to do its thing.